src/java/fr/paris/lutece/plugins/oauth2/dataclient/AbstractDataClient.java

```diff
@@ -250,6 +263,12 @@ public String getData( Token token )
         {
             RequestAuthenticator authenticator = new BearerTokenAuthenticator( token.getAccessToken( ) );
             strResponse = httpAccess.doGet( strUrl, authenticator, null );
+            
+            if ( StringUtils.isNotEmpty( strResponse ) &&_jWTParser != null && _bEnableJwtParser )
+            {                      
+                strResponse = parseJWTResponse( strResponse );
+            }
```

Comment:
You are missing a check. https://openid.net/specs/openid-connect-core-1_0.html#rfc.section.5.3.2 says 

> The sub (subject) Claim MUST always be returned in the UserInfo Response.
>NOTE: Due to the possibility of token substitution attacks (see [Section 16.11](https://openid.net/specs/openid-connect-core-1_0.html#TokenSubstitution)), the UserInfo Response is not guaranteed to be about the End-User identified by the sub (subject) element of the ID Token. The sub Claim in the UserInfo Response MUST be verified to exactly match the sub Claim in the ID Token; if they do not match, the UserInfo Response values MUST NOT be used.